gitclone/git.go

```diff
@@ -363,7 +363,7 @@ func checkout(gitCmd git.Git, arg, branch string, depth int, isTag bool) *step.E
 		if isTag {
 			opts = append(opts, "--tags")
 		}
-		if branch != "" {
+		if branch == arg || (branch != "" && isTag) {
```

Comment:
Can you add a "test" case for when both branch and tag is specified? Why does it differ from the commit one?

Reply:
Yes